lib/OperatingSystems/DefaultOs.php

```diff
@@ -141,50 +146,52 @@ public function getNetworkInfo(): array {
 	}
 
 	public function getNetworkInterfaces(): array {
-		$interfaces = glob('/sys/class/net/*') ?: [];
-		$result = [];
+		$data = [];
 
-		foreach ($interfaces as $interface) {
-			$iface = [];
-			$iface['interface'] = basename($interface);
-			$iface['mac'] = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "link/ether " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
-			$iface['ipv4'] = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "inet " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
-			$iface['ipv6'] = shell_exec('ip -o -6 addr show ' . $iface['interface'] . ' | sed -e \'s/^.*inet6 \([^ ]\+\).*/\1/\'');
-			if ($iface['interface'] !== 'lo') {
-				$iface['status'] = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/operstate');
-				$iface['speed'] = (int)shell_exec('cat /sys/class/net/' . $iface['interface'] . '/speed');
-				if (isset($iface['speed']) && $iface['speed'] > 0) {
-					if ($iface['speed'] >= 1000) {
-						$iface['speed'] = $iface['speed'] / 1000 . ' Gbps';
-					} else {
-						$iface['speed'] = $iface['speed'] . ' Mbps';
-					}
-				} else {
-					$iface['speed'] = 'unknown';
+		foreach ($this->getNetInterfaces() as $interfaceName => $interface) {
+			$netInterface = new NetInterface($interfaceName, $interface['up']);
+			$data[] = $netInterface;
+
+			foreach ($interface['unicast'] as $unicast) {
+				if ($unicast['family'] === self::AF_INET) {
+					$netInterface->addIPv4($unicast['address']);
 				}
-				$duplex = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/duplex');
-				if (isset($duplex) && $duplex !== '') {
-					$iface['duplex'] = 'Duplex: ' . $duplex;
+				if ($unicast['family'] === self::AF_INET6) {
+					$netInterface->addIPv6($unicast['address']);
+				}
+			}
+
+			if ($netInterface->isLoopback()) {
+				continue;
+			}
+
+			$interfacePath = '/sys/class/net/' . $interfaceName;
+
+			try {
+				$netInterface->setMAC($this->readContent($interfacePath . '/address'));
+
+				$speed = (int)$this->readContent($interfacePath . '/speed');
+				if ($speed >= 1000) {
+					$netInterface->setSpeed($speed / 1000 . ' Gbps');
```

Comment:
How is translation handled for these units? Is it?

Reply:
Gbps and Mpbs are not translated.
I guess that's okay for most people as noone complained afaik ;)

Reply:
Not ideal, but that was already like that before the PR?